### ISSUE TITLE
[Suggestion/Request] - A few coordinate based tweaks

### ISSUE THREAD

**Issue by DemonSlayer112** [maruohon/tweakeroo#401]
Hello!
By far this is my favorite mod, it offers so much that are not only unique but also what some other QOL mods do so I can trim down the mod number with this mod! Thank you :)

But I have a few ideas for tweaks that feel like they could potentially blend nicely.

-----

**(1)** <ins>Absolute Position</ins>
This would be a way to get **exactly** on the coordinate you want of the block your on.
Input a number somehow, and the player will be shifted to that coordinate(current block)
this way we can easily align ourselves in the center of a block.

-----

**(2)** <ins>Absolute Rotation</ins>
Same as position, but for where we're facing.

The one I'm reaaaly hopeful for.

-----

**(3)** <ins>Boat turn fix</ins>
You know how piston bolts are used mostly for nether travel?
I like them, but I always worried how much strain they might add.
But we cant use boats!
When we place them down they for whatever reason, rotate.

It would be great to force boats to use the rotation value of the player that placed them.
As well as to be able to prevent another player(or just ourselves) from rotating them after.
I don't know why its so hard to have a boat face 60 degrees when I place one facing 60 degrees and I hope this mod this fix this.
(Also Absolute Rotation would go fantastically with this right?! :D)

**Comment by maruohon** [comment]
I now added a malilib action to set the player's fractional X and Z positions. There have already been actions to set the player yaw and pitch rotations for a few months now in the malilib development code in 1.12.2.

As for 3: I'm pretty sure it's not possible to fix or set the boat rotation from a client mod, as it's not even directly tied to the player rotation is it? And doesn't the boat always face directly along a given axis when placed? TBH I really don't know without going to check... I so rarely have even time to play the game anymore :(

So basically these will arrive in something like MC 1.19 or 1.20 or whatever whenever I finally manage to port all the latest development code from 1.12.2 to the later MC versions.

**Comment by DemonSlayer112** [comment]
Ty for the time! Sorry im not to sure how to respond xD

But i have forgotten a idea!
A tweak/yeet to disable item pickup, or even better, itemPickupRestriction, this way we can control what items we pickup.
You know when your in survival not wanting a dirt inventory from clipping a farm, or just not voiding everything in creative.
A second creative void pickup toggle would also be handy. We can can choose to pick up items in creative and not void them if full.

**Comment by maruohon** [comment]
Any such pickup tweaks would only work in single player if they are added via Tweakeroo, as those need to happen on the server thread. So Tweakeroo, being a client mod, can only affect the server thread while you are playing single player.
